activemodel/test/cases/dirty_test.rb

```diff
@@ -237,4 +238,9 @@ def save
   test "model can be dup-ed without Attributes" do
     assert @model.dup
   end
+
+  test "to_json should work on model" do
+   @model.name = "Dmitry"
+   assert_equal @model.to_json, "{\"name\":\"Dmitry\",\"color\":null,\"size\":null,\"status\":\"initialized\"}"
```

Comment:
@anilmaurya How does this test fail before the patch?

Reply:
Test fails with error

```
/home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:174:in `as_json'
    /home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:58:in `as_json'
    /home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:175:in `block in as_json'
    /home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:174:in `each'
    /home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:174:in `as_json'
    /home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:58:in `as_json'
    /home/vagrant/rails/activesupport/lib/active_support/json/encoding.rb:35:in `encode'
    /home/vagrant/rails/activesupport/lib/active_support/json/encoding.rb:22:in `encode'
    /home/vagrant/rails/activesupport/lib/active_support/core_ext/object/json.rb:43:in `to_json'
    /home/vagrant/rails/activemodel/test/cases/dirty_test.rb:244:in `block in <class:DirtyTest>'
```